src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminAreasOfLawService.java

```diff
@@ -53,6 +56,16 @@ public AreaOfLaw createAreaOfLaw(final AreaOfLaw areaOfLaw) {
         return new AreaOfLaw(areasOfLawRepository.save(createNewAreaOfLawEntityFromModel(areaOfLaw)));
     }
 
+    public void deleteAreaOfLaw(final Integer areaOfLawId) {
+        try {
+            areasOfLawRepository.deleteById(areaOfLawId);
+        } catch (EmptyResultDataAccessException ex) {
+            throw new NotFoundException(ex);
+        } catch (DataAccessException ex) {
```

Comment:
When there is a foreign key constraint violation (trying to delete an AoL that is referred to in the CourtAreaOfLaw table), I expected a DataIntegrityViolationException to be thrown but it isn't. I'm struggling to find a subclass of DataAccessException that we can catch rather than using this base exception.

Reply:
As we said in the dev-standup. Might be best to make a method that has a query to determine how many rows would be affected. If it is one or more, then throw a custom exception to denote the issue. We can then pick that up on the front end and go from there :- )